src/analytics.ts

```diff
@@ -89,9 +97,12 @@ export function createAnalyticsInstance(options?: Options) {
 
     const trackEvent = <T extends keyof TEvents>(event: T, analyticsData: TEvents[T]) => {
         if (!_rudderstack) return
-
-        _rudderstack?.track(event, { ...coreData, ...analyticsData })
+        if (navigator?.onLine) {
+            console.log(event, { ...coreData, ...analyticsData })
+            _rudderstack?.track(event, { ...coreData, ...analyticsData })
+        }
     }
```

Comment:
What's the use case of checking the online status of the browser here ? Let's say if someone called the track event while he was offline, but now he is back online are we accumulating those events to be able to send those to the rudderstack server later ?

Reply:
If user is offline, it means that he/she can't interact with react application properly. Exclude navigation change of course
To solve that we have to make some kind of cache and accumulate. And once user become online push them once to tracking system. So for that we need observer, but we not in react app. 
looks like `addEventlistener(''offline' () => {})` can't help us

I have to say that it's too difficult to me

Reply:
What is the goal here? Do we want to trace actions even if the users gone offline?

Reply:
I assume yes @prince-deriv 
It's tricky to achieve

Reply:
Resolved as `offline_cache` under `trackEvent` function